tests/chainer_tests/functions_tests/loss_tests/test_crf1d.py

```diff
@@ -1,90 +1,119 @@
+import itertools
 import unittest
 
 import numpy
 
 import chainer
+from chainer import cuda
 from chainer import functions
+from chainer import gradient_check
 from chainer import testing
+from chainer.testing import attr
 
 
+@testing.parameterize(
+    {'lengths': [3, 3], 'batches': [2, 2, 2]},
+    {'lengths': [3, 2], 'batches': [2, 2, 1]},
```

Comment:
I add two test cases
- batches = [2] (length of sequence is one)
- batches = [3, 1, 1]

I feel it's enough because they cover all branches in the code.